### PR TITLE
Add ability for interface binding

### LINKED ISSUE #61: Add option to bind speedtest to run on specific interface "--source SOURCE"
where SOURCE = eth0/wlan0/etc. or  IP address

We are contemplating migrating an existing project to nodejs where we'd need to be able to run the speedtest over multiple different interfaces (basically checking speedtest on wlan0 and eth0)

Similar to functionality on https://github.com/sivel/speedtest-cli

It would be preferable to be able to specify interface name (eth0) instead of the IP address (could use something like https://www.npmjs.com/package/network)

```
var speedTest = require('speedtest-net');
var test = speedTest({maxTime: 5000, source: "eth0"});
```

Should include interface in output also `{interface: "eth0"}` or `{interface: "192.168.1.10"}`

All tests should pass.

### PR DESCRIPTION
**Pull request by DjLegolas**
Can now use `--sourceip IP-ADDRESS` arg with the bin file
Or use the `options` object with the key `sourceIp`

It works with IP addresses only.

Closes #61.

### CODE REVIEW

File: bin/index.js [hunk 1]
```diff
@@ -170,23 +170,30 @@ updateLines();
...
+};
 
 /*
  * get parameters
  */
 var errorParameter = null;
 process.argv.forEach(function (val, index, array) {
   if (index >= 2) {
-    if (val.startsWith("--proxy")) {
+    if (val.startsWith('--proxy')) {
```
Comment (ddsol) [inline bin/index.js]: Could you change the `startsWith` to `val === '--proxy'`? Not sure why `--proxyficationatorial` should be considered valid. Ditto below. Also, would you mind changing the `!= undefined` to `!== undefined` (twice)?

Comment (DjLegolas) [reply bin/index.js]: A late response, bit I fixed this..

**Comment by ddsol** [comment]
Superseded by #95 (rebase)

**Comment by rodcul** [comment]
hey @DjLegolas I just tested this and the flag seems to be ignored. Any ideas?

image

**Comment by DjLegolas** [comment]
Hi @rodcul, this option will let you choose a specific network interface based on it's IP address.
If there is no interface with the given IP, it will not work.

This option will not spoof the source IP.

**Pull request closed without merging**